### PR TITLE
Enhance camera transitions

### PR DESCRIPTION
**Pull request by lebull**
[](http://www.youtube.com/watch?v=-sUcBKfSETY "")

This PR implements transition durations as well as offers a selection of different transition curves.  For this PR, I tried to not touch existing code as much as possible.

TODO:

- [ ] Reimplement transition type and set that as default to preserve previous behaviors
 * Right now, I am only able to approximate it, and I'm having trouble reading the original code.
 
- [x] Add Transition Types to the settings file
- [x] Doublecheck that I didn't do anything stupid 

Issues:

* Not an actual bug, but the existing behavior of exiting an orbit is sometimes VERY jarring when using .
* I feel like in my video, my movements affected the camera's current position a lot, and I think it just needs to be the lookat if I'm not mistaken.  Needs to ask about this/look.

### CODE REVIEW

File: Src/Camera/CameraData.cs [hunk 1]
```diff
@@ -120,6 +124,7 @@ public static void LoadSettings()
                 ws.WriteLine("	LookAt={x='0.0', y='-0.5', z='1.0'} ");
                 ws.WriteLine("	MinTime='4.0' ");
                 ws.WriteLine("	MaxTime='8.0' ");
+                ws.WriteLine("  TransitionTime='1.0'");
```
Comment (lebull) [inline Src/Camera/CameraData.cs:127]: Replace with tabs

Comment (lebull) [reply Src/Camera/CameraData.cs:127]: Also, do we even want to change the default settings?

File: Src/MyCameraPlugin.cs [hunk 2]
```diff
@@ -466,5 +367,164 @@ void BlendCameraPose()
     public void OnDestroy() {
 
     }
-}
 
+   private void transitionToNextCamera(bool transitionToMenu)
```
Comment (friesofdoom) [inline Src/MyCameraPlugin.cs:371]: I'd rather all function names start with an upper case letter, for consistency

**Comment by friesofdoom** [comment]
>     * I feel like in my video, my movements affected the camera's current position a lot, and I think it just needs to be the lookat if I'm not mistaken.  Needs to ask about this/look.

Yes, this is when the look-at is a bone on the player. The code applies (or used to) exponential smoothing to this position. The lookat position smoothing needs to be cranked up a bit.

File: Src/Camera/CameraData.cs (hunk 1 above)
Comment (friesofdoom) [reply Src/Camera/CameraData.cs:127]: I've got no problem with that, but we might want to put in more error checking if they values are not found in the loaded settings file.

**Comment by friesofdoom** [comment]
Overall I like the refactoring, the orbital transition still scares me though 😓

**Review by friesofdoom — APPROVED** [review]

File: Src/Camera/CameraData.cs (hunk 1 above)
Comment (sepffuzzball) [reply Src/Camera/CameraData.cs:127]: I think that would also be helpful! I think overall some settings changes could be in order to make things easier too~

**Comment by sepffuzzball** [comment]
Looks good to me!

**Review by sepffuzzball — APPROVED** [review]

**Pull request merged**